### DISCUSSION TITLE
Maximum path length?

### DISCUSSION THREAD

**Discussion by Rastalovich** [gboudreau/Greyhole#257] [Q&A] [answered]
Evening,

I'm transferring one Gh setup to another and I've noticed a curious thing. The landing zone(s) are a lot bigger than I'd have guessed.
So I looked into finding the largest files within a directory, nothing precise, just found a rough estimator using:

sudo du -a /mnt/GHLZ 2>/dev/null | sort -n -r | head -n 20

EDIT: actually I found a better way, if you use this command, if all links are done correctly, you should get no output:
------------------------------------------------------------------------
find /mnt/GHLZ -type f -iname "*.*" -printf '%s %p\n'| sort -nr | head -10

(on my source /mnt/GHLZ I get a long list (increasing the head -10) of files with a huge path name) for example:

123691297 /mnt/GHLZ/Cinema/Cinema/Anime Manga/FIST of the NORTH STAR - Complete 1984-2015 Collection - 480p-720p x264/b. Series (480p-720p)/a. Fist of the North Star (Japanese Dub - 480p)/Season 1 (1984-87)/Chapter 1 - The SHIN Arc (Eps. 01-22)/Fist of the North Star (HnK) - S01 A01 E021 - The Palace of Evil in Flames - Shin.mp4
-------------------------------------------------------------------------

And it appears that there are files that have remained in the landing zone and not transferred to the pool, and no link created. Integrity wise, this is ok cos the file still exists, however, I do tend to use an SSD and allocate it all to a landing zone. Generally 100g or so I've found to be ok.

The source landing zone is however at 40g used out of 99g. And whilst I know for a fact I abuse the principles of Gh, there are 100s of 1000s of files in the pool, I'd expect just links to be a heck of alot smaller than that.

The destination landing zone I noticed is behaving similarly, and I'm merely using the cp command (cp -vR) to copy from one to the other - this tells me the path is going to be the same length.

So I wonder, is there a limit to path length? It's not a problem if it can't be changed, I can just move the files up a level or whatever.

**Comment by gboudreau** [comment, marked as answer]
The max is from Samba (and CIFS mounts, when using mount_shares_locally) : 255 characters
Ref: https://serverfault.com/questions/393922/is-there-a-maximum-filepath-length-over-samba-and-can-it-be-changed#:~:text=1%20Answer&text=There%20has%20been%20a%20limit,%2Dfile%2Dpath%2Dlimit%20.

The path of the files in the LZ would also be limited to 255 characters I think.

**Reply by Rastalovich** [reply]
Makes sense in that case. It's not a big issue because of course the files are still intact. I'll look to rectify by copying out and back in once I've changed the path/names.

**Reply by Rastalovich** [reply]
I've cut the folder names down and they're transferring fine.
Some other directories I've noticed are taking up space in the GHLZ, a theme being some Mac backups I've done where although they're 'apps' on the Mac, of course they're containers with folders inside them - big pathnames with them so I'll likely zip them up.

Be an interesting inclusion to the graphical interface to have the landing-zone searched for 'non-transferred-files'.
